src/main/java/net/minecraftforge/fluids/FluidAttributes.java

```diff
@@ -458,4 +444,36 @@ public FluidAttributes build()
             return new FluidAttributes(this);
         }
     }
+
+    public static class Water extends FluidAttributes
+    {
+        protected Water(Builder builder)
+        {
+            super(builder);
+        }
+
+        @Override
+        public int getColor(IEnviromentBlockReader world, BlockPos pos)
+        {
+            return BiomeColors.getWaterColor(world, pos);
+        }
+
+        public static FluidAttributes.Builder builder(Fluid owner, ResourceLocation stillTexture, ResourceLocation flowingTexture) {
+            return new Builder(owner, stillTexture, flowingTexture);
+        }
+
+        private static class Builder extends FluidAttributes.Builder
+        {
+            protected Builder(Fluid owner, ResourceLocation stillTexture, ResourceLocation flowingTexture)
+            {
+                super(owner, stillTexture, flowingTexture);
+            }
+
+            @Override
+            public FluidAttributes build()
+            {
+                return new Water(this);
```

Comment:
Could we not add a simpler way of subclassing the attributes without subclassing the builder. e.g. by optionally passing in a factory to `build` (i.e. the water builder would become `build(Water::new)`) or to the builder constructor, or adding a new builder function `setAttributesClass` or something.
That would make it easier for other mods which wanted to add a custom colour map, or otherwise override one of the FluidAttributes implementations.

Reply:
This should be a simple enough change to add, you should do it.